File: incomplete_fizz_buzz/go/test/src/helper/printer_spy.go
package helper

type PrinterSpy struct {
  strings []string
}

func CreatePrinterSpy() *PrinterSpy {
  instance := new(PrinterSpy)
  instance.strings = []string{}
  return instance
}

func (self *PrinterSpy) Execute(s string) {
  self.strings = append(self.strings, s)
}

func (self *PrinterSpy) Results() []string {
  return self.strings
}

File: incomplete_fizz_buzz/java/test/PrinterSpy.java
import output.Printable;

import java.util.List;
import java.util.ArrayList;

public class PrinterSpy implements Printable {

  private List<String> strings = new ArrayList<String>();

  @Override
  public void execute(String s) {
    strings.add(s);
  }

  public List<String> results() {
    return strings;
  }
}
